os/board/rtl8730e/src/rtl8730e_ist415.c

```diff
@@ -0,0 +1,181 @@
+static void rtl8730e_ist415_irq_attach(touch_handler_t handler, FAR char *arg);
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+static struct rtl8730e_ist415_touchinfo_s g_ist415info = {
+	.touch_config = {
+		.i2c_config = {
+			.frequency = IST415_I2C_FREQ,
+			.address = IST415_I2C_ADDR,
+			.addrlen = IST415_I2C_ADDRLEN,
+		},
+		.attach = rtl8730e_ist415_irq_attach,
+		.irq_enable = rtl8730e_ist415_enable_irq,
+		.irq_disable = rtl8730e_ist415_disable_irq,
+	},
+	.handler = NULL,
+};
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+static void rtl8730e_ist415_irq_handler(uint32_t id, gpio_irq_event event)
+{
+	/* we are using level based interrupt trigger here,
+	 * so, we have to disable this particular gpio interrupt
+	 * until we finish this particular interrupt related work
+	 * in the HPWORK thread
+	 */
+	gpio_irq_disable(&g_ist415info.data_ready);
+
+	if (g_ist415info.handler != NULL) {
+		g_ist415info.handler((void *)id);
+	} else {
+		gpio_irq_enable(&g_ist415info.data_ready);
+	}
+}
+
+static void rtl8730e_ist415_enable_irq(void)
+{
+	gpio_irq_enable(&g_ist415info.data_ready);
+}
+
+static void rtl8730e_ist415_disable_irq(void)
+{
+	gpio_irq_disable(&g_ist415info.data_ready);
+}
+
+static void rtl8730e_ist415_irq_attach(touch_handler_t handler, FAR char *arg)
+{
+	g_ist415info.handler = handler;
+	gpio_irq_init(&g_ist415info.data_ready, IST415_GPIO_I2C_PIN, rtl8730e_ist415_irq_handler, (uint32_t)arg);
+	gpio_irq_set(&g_ist415info.data_ready, IRQ_FALL_RISE, 1);
+	gpio_irq_enable(&g_ist415info.data_ready);
+}
+
+static void rtl8730e_ist415_gpio_reset(bool flag)
+{
+	if (!flag) {
+		GPIO_WriteBit(IST415_GPIO_RESET_PIN, PIN_LOW);
+	} else {
+		GPIO_WriteBit(IST415_GPIO_RESET_PIN, PIN_HIGH);
+	}
+	DelayMs(500);
+	return;
+}
+
+static void rtl8730e_ist415_gpio_init(void)
+{
+	Pinmux_Config(IST415_GPIO_RESET_PIN, PINMUX_FUNCTION_GPIO);
+	GPIO_InitTypeDef TouchResetPin;
+	TouchResetPin.GPIO_Pin = IST415_GPIO_RESET_PIN;
+	TouchResetPin.GPIO_PuPd = GPIO_PuPd_NOPULL;
+	TouchResetPin.GPIO_Mode = GPIO_Mode_OUT;
+	GPIO_Init(&TouchResetPin);
+}
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+void rtl8730e_ist415_initialize(void)
+{
+	FAR struct i2c_dev_s *i2c;
+	
+	rtl8730e_ist415_gpio_init();
+	rtl8730e_ist415_gpio_reset(false);
+	i2c = up_i2cinitialize(IST415_I2C_PORT);
+	if (!i2c) {
+		touchdbg("ERROR: Failed to initialize I2C\n");
+		return;
+	}
+	int ret = 0;
+	uint8_t reg[1];
+	reg[0] = 0x23;
+	ret = i2c_write(i2c, &(g_ist415info.touch_config.i2c_config), (uint8_t *)reg, 1);
+	if (ret < OK) {
+		touchdbg("ERROR: i2c_write failed\n");
+	}
```

Comment:
This does not seem to be a board-specific code.
Please move to ist415.c

Reply:
one write is necessary for working. without this, touch won't work.
I have moved other write and read to driver.